src/mono/mono/mini/mini-wasm-debugger.c

```diff
@@ -456,6 +460,30 @@ get_object_id(MonoObject *obj)
 	return ref->id;
 }
 
+static void
+assembly_loaded (MonoProfiler *prof, MonoAssembly *assembly)
+{
+	DEBUG_PRINTF (2, "assembly_loaded callback called for %s\n", assembly->aname.name);
+	MonoImage *assembly_image = assembly->image;
+	MonoImage *pdb_image = NULL;
+	MonoDebugHandle *handle = mono_debug_get_handle (assembly_image);
+	if (handle) {
+		MonoPPDBFile *ppdb = handle->ppdb;
+		if (!mono_ppdb_is_embedded (ppdb)) //if it's an embedded pdb we don't need to send pdb extrated to DebuggerProxy.
+			pdb_image = mono_ppdb_get_image (ppdb); 
+	}
+
+	if (pdb_image) {
+		mono_wasm_asm_loaded (
+			assembly_image->assembly_name, assembly_image->raw_data, assembly_image->raw_data_len,
+			pdb_image->raw_data, pdb_image->raw_data_len);
+	} else if (mono_has_pdb_checksum (assembly_image->raw_data, assembly_image->raw_data_len)) { //if it's a release assembly we don't need to send to DebuggerProxy
```

Comment:
Is it always the case that `pdb_image` will be NULL for files that have a checksum?  Would it make sense to check for the checksum first?

Reply:
Yes, the case where we are doing an assembly load of some file that we don't have the pdb locally and it's not embedded either, but could have the pdb in a symbol server for example?

Reply:
I will change the order of the checks.

Reply:
So, just for my understanding, is this what you mean?

```
if (!mono_has_pdb_checksum) return; // nothing to be sent

if (pdb_image) { .. send asm + pdb .. }
else { .. send just asm.. }
```

Reply:
oh I think I misunderstood the code and what a checksum is.

I thought that there is a checksum present only on "official" released images for which we can expect to find the pdb on a symbol server.

What I was thinking:
```
  if (mono_has_pdb_checksum) { send asm /* don't send pdb, can get it from a symbol server */ }
  else if (pdb_image) { send asm + pdb /* no pdb on symbol server, but we have it in the budle */ }
  else { send asm only; /* no pdb on server or in bundle, just send the assembly */ }
```

Reply:
I rewrote the code, please check if it's clearer?

Reply:
if (!mono_has_pdb_checksum) return; // nothing to be sent

if (pdb_image && !ppdb_embedded) { .. send asm + pdb .. }
else { .. send just asm.. }